### DISCUSSION TITLE
Chunks are larger than what it is set to.

### DISCUSSION THREAD

**Discussion by getsetbro** [rpldy/react-uploady#121] [Q&A] [answered]
In am using ChunkedUploady with a chunksize set to 350000. The first chunk that is uploaded is set to that size but after that chunks are larger as reported in the useChunkStartListener > data.chunkItem.file.size.

```
useChunkStartListener((data) => {
  return { sendOptions: { method: "POST", paramName: "data",
      params: {
        fileSize: data.chunkItem.file.size,
      },
    },
  };
});
```

Could this have any thing to do with what the server responds with or does it just work with the file size and chunk id?

**Comment by yoavniran** [comment]
Hey @getsetbro, interesting question.
ChunkedUploady (using chunked-sender) uses the browser's ability to slice blobs. The number passed to slice is the number defined by the chunkSize param.
However, I believe it depends on the internal browser implementation and may have slightly different outcomes. Might also be influenced by the file's encoding

What is the difference you're seeing? Is it big or a few bytes?
Is it the same in a different browser?
Is it causing an issue?

**Comment by getsetbro** [comment, marked as answer]
My team figured this out. The chunk size was coming in via a REACT_APP_ environment variable - which is a string, not a number. The JS was then adding the strings instead numbers. Converting the string to a number first solves it.
